### ISSUE TITLE
no-restyle misclassifies flex-grow and @utility classes, and tells you to fix the spelling

### ISSUE THREAD

**Issue by justinchen** [shadcn-ui/lint#18]
`no-restyle` classifies some real utilities as unknown, and the message tells the author to fix a spelling that is correct.

### 1. `flex-grow`

```tsx
<IconButton className="flex-grow" />
```

reports:

```
"flex-grow" restyles <IconButton> (unclassified) — the grammar does not recognize it. Fix the spelling.
```

Tailwind 4.3 does generate `.flex-grow { flex-grow: 1 }` (verified via `compile().build()`), and `shadcn/no-unknown-classes` agrees it is valid. It is also a layout utility, so `allow: ["layout"]` should permit it.

### 2. Project utilities declared with `@utility`

```css
@utility max-w-chat {
  max-width: 48rem;
}
```

```tsx
<Alert className="max-w-chat" />
```

`no-unknown-classes` knows this class; `no-restyle` calls it unclassified and flags it. It is an outer width constraint, i.e. placement.

### Expected

- `no-restyle` classifies every utility the compiler generates, including `@utility` declarations.
- The unknown-class message does not assert a typo for a class that compiles. For agents in particular, "fix the spelling" is acted on literally.

Versions: `@shadcn/lint` 0.1.0, oxlint 1.83, Tailwind 4.3.

**Comment by Sebastian-Sole** [comment]
#52 will not close this one. It reads `@theme` blocks only, by design, so `@utility` classes stay where they are. On `50221d7`, `text-md-regular` is still classified as a color and `heading-2xs` is still unclassified, same as on `main`.

The plugin already knows the class exists. `declaresUtility` in `packages/lint/src/project/theme.ts` is what produces the "your CSS declares it, and the grammar cannot tell what it changes" message. What is missing is a step from the utility's declaration body to a cn group, so a `@utility` that sets `font-size` lands in typography instead of nowhere. That is a CSS-property to group table, which is a good deal more than a redirect, so I am not sending a PR for it.

What it costs in practice: our generated design system puts every text style behind `@utility`, so `text-md-regular`, `heading-2xs` and `display-big` are all invisible to `no-restyle`. The rule still rejects them, because we allow nothing on a component, so the verdict is right either way. The message is what is wrong. It tells you to fix the spelling of a class that is spelled correctly.

**Comment by shadcn** [comment]
@Sebastian-Sole Taking a look at this.
